src/dragtor/utils/audio_utils.py

```diff
@@ -0,0 +1,67 @@
+import subprocess
+import requests
+import shutil
+import re
+
+from loguru import logger
+from pathlib import Path
+from dragtor import config
+
+
+def preprocess_audio_file(audio_path: str) -> Path:
+    """
+    Prepares audio file into .wav format to match the model requirement.
+
+    Returns:
+        A path where the .wav file is temporarily cached.
+    """
+    output_dir = Path(config.conf.audio.files)
+    output_dir.mkdir(exist_ok=True)
+
+    logger.info(f"Processing: {audio_path}")
+    if audio_path.startswith(("http://", "https://")):
+        inputs = requests.get(audio_path).content
+        file_path = output_dir / "_".join(Path(audio_path).parts[-2:])
+        file_path.write_bytes(inputs)
+        logger.debug(f"File downloaded and saved at {file_path}")
+        file_path = convert_to_wav(file_path)
+    else:
+        file_path = convert_to_wav(Path(audio_path))
+
+    return file_path
+
+
+def convert_to_wav(audio_path: Path) -> Path:
+    """Converts non-wav audio file into .wav file."""
+    if audio_path.suffix.lower() != ".wav":
+        wav_file = audio_path.with_suffix(".wav")
+        command = ["ffmpeg", "-y", "-i", str(audio_path), "-ar", "16000", str(wav_file)]
+        subprocess.run(command, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        return wav_file
+    return audio_path
+
+
+def parse_transcript(transcript: str) -> str:
```

Comment:
extremely unclear what is the difference between `parse_transcript`, `parse_audio_transcript` and `parse_diarization`